adjustText/adjustText.py

```diff
@@ -529,6 +544,8 @@ def adjust_text(texts, x=None, y=None, add_objects=None, ax=None,
         qx = np.sum([i[0] for i in [q1, q2, q3]])
         qy = np.sum([i[1] for i in [q1, q2, q3]])
         histm = np.mean(np.array(history), axis=0)
+        # Stop if we've reached the precision threshold, or if the x and y displacement
+        # are both greater than the average over the last 10 iterations (why?)
```

Comment:
To the question why - to prevent spending forever on plots where the algorithm doesn't converge. Why it's mean over max, for example, I don't know, maybe that would be more logical...

Reply:
Ah, okay. I think this would definitely make sense if `q` represented the total 'badness' of the current arrangement, but since it represents something like the total distance moved by the text, it's not so clear. I guess ideally the text would jump around a lot initially, and taper to smaller and smaller adjustments with each iteration, so if the amount of movement isn't decreasing over time, that might suggest it's spinning its wheels?

Reply:
Yep, that's exactly the idea - and I think it is indeed working like that in general.